src/main/java/uk/gov/hmcts/reform/judicialbooking/domain/service/common/PersistenceService.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.judicialbooking.domain.service.common;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingEntity;
+import uk.gov.hmcts.reform.judicialbooking.data.BookingRepository;
+import uk.gov.hmcts.reform.judicialbooking.domain.model.Booking;
+import uk.gov.hmcts.reform.judicialbooking.util.PersistenceUtil;
+
+@Service
+public class PersistenceService {
+    private BookingRepository bookingRepository;
+    private PersistenceUtil persistenceUtil;
+
+    public PersistenceService(BookingRepository bookingRepository, PersistenceUtil persistenceUtil) {
+        this.bookingRepository = bookingRepository;
+        this.persistenceUtil = persistenceUtil;
+    }
+
+    @Transactional
```

Comment:
The transactional annotation should be at the controller method directly. we can keep it here as well.